### PR TITLE
[9413] Conch documentation fixes

### PR DESCRIPTION
**Pull request by wiml**
* [X] There is an associated ticket in Trac: https://twistedmatrix.com/trac/ticket/9413
* [X] The changes pass minimal style checks
* [X] I have created a newsfragment
* [X] No changes to behavior or the automated tests.

### CODE REVIEW

**Comment by wiml** [comment]
This PR doesn't change any code (only docstrings), so I think the codecov failure is spurious.

File: src/twisted/conch/interfaces.py [hunk 1]
```diff
@@ -18,30 +18,36 @@ class IConchUser(Interface):
...
         We return a subclass of L{SSHChannel<ssh.channel.SSHChannel>}.  If
         an appropriate channel can not be found, an exception will be
-        raised.  If a L{ConchError<error.ConchError>} is raised, the .value
-        will be the message, and the .data will be the error code.
+        raised.  If a L{ConchError<error.ConchError>} is raised, the C{.value}
+        will be the message, and the C{.data} will be the error code.
 
-        @type channelType:  L{str}
+        @param channelType: The requested channel type
+        @type channelType:  L{bytes}
```
Comment (markrwilliams) [inline src/twisted/conch/interfaces.py]: Whoops!  This certainly [appears to be bytes](https://github.com/twisted/twisted/blob/91a72d075db94a78da07d8d601fad090d91d5504/src/twisted/conch/ssh/connection.py#L136).  Thanks for fixing the type here!

File: src/twisted/conch/interfaces.py [hunk 2]
```diff
@@ -18,30 +18,36 @@ class IConchUser(Interface):
...
         """
         The other side requested a subsystem.
-        subsystem is the name of the subsystem being requested.
-        data is any other packet data (often nothing).
 
-        We return a L{Protocol}.
+        @param subsystem: The name of the subsystem being requested
+        @type subsystem: L{bytes}
+        @param data:     Additional request data (often nothing)
+        @type data:      L{bytes}
```
Comment (markrwilliams) [inline src/twisted/conch/interfaces.py]: :+1: This is [patently binary data](https://github.com/twisted/twisted/blob/91a72d075db94a78da07d8d601fad090d91d5504/src/twisted/conch/ssh/session.py#L40).

File: src/twisted/conch/interfaces.py [hunk 3]
```diff
@@ -18,30 +18,36 @@ class IConchUser(Interface):
...
+        @param channelType: The requested channel type
+        @type channelType:  L{bytes}
+        @param windowSize:  The initial size of the remote window
         @type windowSize:   L{int}
+        @param maxPacket:   The largest packet we should send
         @type maxPacket:    L{int}
-        @type data:         L{str}
-        @rtype:             subclass of L{SSHChannel}/L{tuple}
+        @param data:        Additional request data
+        @type data:         L{bytes}
```
Comment (markrwilliams) [inline src/twisted/conch/interfaces.py:40]: :+1: Also [binary data](https://github.com/twisted/twisted/blob/trunk/src/twisted/conch/ssh/connection.py#L548) (from [`ssh_CHANNEL_OPEN`](https://github.com/twisted/twisted/blob/trunk/src/twisted/conch/ssh/connection.py#L123-L141)).

File: src/twisted/conch/interfaces.py [hunk 4]
```diff
@@ -18,30 +18,36 @@ class IConchUser(Interface):
...
 
     def lookupSubsystem(subsystem, data):
         """
         The other side requested a subsystem.
-        subsystem is the name of the subsystem being requested.
-        data is any other packet data (often nothing).
 
-        We return a L{Protocol}.
+        @param subsystem: The name of the subsystem being requested
+        @type subsystem: L{bytes}
```
Comment (markrwilliams) [inline src/twisted/conch/interfaces.py]: :+1: [Binary data](https://github.com/twisted/twisted/blob/91a72d075db94a78da07d8d601fad090d91d5504/src/twisted/conch/ssh/session.py#L38)!

File: src/twisted/conch/interfaces.py [hunk 5]
```diff
@@ -18,30 +18,36 @@ class IConchUser(Interface):
...
+        @param data:     Additional request data (often nothing)
+        @type data:      L{bytes}
+        @rtype:          L{Protocol} or L{None}
```
Comment (markrwilliams) [inline src/twisted/conch/interfaces.py:55]: When can this be `None`?

File: src/twisted/conch/interfaces.py [hunk 6]
```diff
@@ -18,30 +18,36 @@ class IConchUser(Interface):
...
+        @type channelType:  L{bytes}
+        @param windowSize:  The initial size of the remote window
         @type windowSize:   L{int}
+        @param maxPacket:   The largest packet we should send
         @type maxPacket:    L{int}
-        @type data:         L{str}
-        @rtype:             subclass of L{SSHChannel}/L{tuple}
+        @param data:        Additional request data
+        @type data:         L{bytes}
+        @rtype:             a subclass of L{SSHChannel} or L{None}
```
Comment (markrwilliams) [inline src/twisted/conch/interfaces.py:41]: When can this be `None`?

**Review by markrwilliams — APPROVED** [review]
Thanks for improving `conch`'s documentation!

I agree that the Codecov failure is spurious.  I've pushed this to a branch so the builders will run on it and this can pass all required checks.

Feel free to address the review comments asking when methods can return `None` in a follow up PR.

Thanks again!

**Comment by wiml** [comment]
I added a few sentences to explain the `None` return values.

**Comment by twm** [comment]
The Fedora buildbot is failing with nonsense. o.O

```
[ERROR]
Traceback (most recent call last):
  File "/buildslave/fedora26-python-3.6/Twisted/build/py36-alldeps-nocov-posix/lib/python3.6/site-packages/twisted/trial/_synctest.py", line 1375, in _runCleanups
    f(*args, **kwargs)
  File "/buildslave/fedora26-python-3.6/Twisted/build/py36-alldeps-nocov-posix/lib/python3.6/site-packages/twisted/internet/test/reactormixins.py", line 225, in unbuildReactor
    reactor._uninstallHandler()
  File "/buildslave/fedora26-python-3.6/Twisted/build/py36-alldeps-nocov-posix/lib/python3.6/site-packages/twisted/internet/posixbase.py", line 324, in _uninstallHandler
    self._childWaker.uninstall()
  File "/buildslave/fedora26-python-3.6/Twisted/build/py36-alldeps-nocov-posix/lib/python3.6/site-packages/twisted/internet/posixbase.py", line 214, in uninstall
    _signals.installHandler(-1)
  File "/buildslave/fedora26-python-3.6/Twisted/build/py36-alldeps-nocov-posix/lib/python3.6/site-packages/twisted/internet/_signals.py", line 54, in installHandler
    signal.signal(signal.SIGCHLD, signal.SIG_DFL)
  File "/usr/lib64/python3.6/signal.py", line 47, in signal
    handler = _signal.signal(_enum_to_int(signalnum), _enum_to_int(handler))
builtins.TypeError: 'int' object is not callable

twisted.internet.test.test_process.ProcessTestsBuilder_EPollReactorTests.test_systemCallUninterruptedByChildExit
```

Clearly nothing to do with this change.

**Pull request merged**